### PR TITLE
CORTX-32741: HARE - Consul Support for GCONF

### PR DESCRIPTION
**Pull request by pavankrishnat**
Problem:
Provisioner team is planning to migrate file based Gconf to consule based Gconf.
In order to do this, hare component needs some changes to support this migration.

Solution:
There are changes required in way we fetch the keys and its values from Gconf
 e.g. endpoints, Hostname, machineIDs.
Approach we took is,
a) for any key we want to fetch, first we need to fetch num_XXX key from Gconf.
b) based on this number, we need to iterate over each key and get desired value of key.

Signed-off-by: pavankrishnat <pavan.k.thunuguntla@seagate.com>

### CODE REVIEW

**Comment by pavankrishnat** [comment]
Build Imaged: https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/7281/console
3N deployment: https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/9379/console
Sanity: https://eos-jenkins.colo.seagate.com/job/Hare/job/Hare_k8s_Sanity_PR/320/console

**Comment by vaibhavparatwar** [comment]
@pavankrishnat  can we test 15N / 27N deployments as well?

**Comment by pavankrishnat** [comment]
> @pavankrishnat can we test 15N / 27N deployments as well?

Yes, we can test

**Comment by pavankrishnat** [comment]
retest this please

File: provisioning/miniprov/hare_mp/cdf.py [hunk 1]
```diff
@@ -148,15 +147,7 @@ def _get_pool_property(self, pool: PoolHandle, prop_name: str) -> int:
...
 
     def _get_layout(self, pool: PoolHandle) -> Optional[Layout]:
-        conf = self.provider
         (cluster_id, pool_type, storage_ndx) = pool.tuple()
-        # cluster>storage_set[N]>durability>{type}
-        type_value = conf.get(
-            f'cluster>storage_set[{storage_ndx}]'
-            f'>durability>{pool_type}',
-            allow_null=True)
-        if not type_value:
```
Comment (Shreya-18) [inline provisioning/miniprov/hare_mp/cdf.py:158]: Are we not using `type_value ` anymore?
I see if the `type_value` for fetching pool type is missing then we directly return None.

File: provisioning/miniprov/hare_mp/cdf.py [hunk 2]
```diff
@@ -312,41 +306,49 @@ def _create_ports_descriptions(self, hostname: str) -> NetworkPorts:
...
                             _hax = round(_parsed_url.port / 100) * 100
             elif srv == 'm0_client_other':
-                for client in self.provider.get_motr_clients():
-                    url = client.get('endpoints')
-                    if url:
-                        for u in url or ():
-                            _parsed_url = urlparse(u)
+                num_clients = int(conf.get('cortx>motr>num_clients'))
+                for i in range(num_clients):
+                    num_endpoints = conf.get(
```
Comment (Shreya-18) [inline provisioning/miniprov/hare_mp/cdf.py:323]: Let's follow the same convention of getting the len of array.
```suggestion
                    num_endpoints = int(conf.get(
```

**Comment by vaibhavparatwar** [comment]
@SwapnilGaonkar7 @d-nayak @Shreya-18 Please review.. @pavankrishnat if you have addressed review comments from @Shreya-18 please resolve the comments or mention so.

File: provisioning/miniprov/hare_mp/store.py [hunk 3]
```diff
@@ -95,7 +92,14 @@ def get_machine_ids_for_service(self, service_type: str) -> List[str]:
         service_type will be like Const.SERVICE_MOTR_IO.value,
         Const.SERVICE_S3_SERVER.value etc
         """
-        return self.get_machine_ids_for_attribute('services', service_type)
+        if service_type == 'io':
+            service_type = 'data_node'
+        elif service_type == 'rgw_s3':
```
[the remark this answers is not in the document]
Comment (Shreya-18) [reply provisioning/miniprov/hare_mp/store.py:97]: Also.. a better way of doing this if else part is via `dictionaries`

**Comment by pavankrishnat** [comment]
> @pavankrishnat can we test 15N / 27N deployments as well?
@SwapnilGaonkar7 added console output [console_output.log](https://github.com/Seagate/cortx-hare/files/9179072/console_output.log)

**Comment by pavankrishnat** [comment]
@mssawant @Shreya-18, I have address all your review comments and suggestions, please let me know, if any.

**Comment by pavankrishnat** [comment]
retest this please

**Comment by pavankrishnat** [comment]
retest this please

**Review by Shreya-18 — APPROVED** [review]

**Comment by vaibhavparatwar** [comment]
@mukul-seagate11  need your help in CLA verification issue here. cc @pavankrishnat @SunilSavanurSeagate

**Review by vaibhavparatwar — APPROVED** [review]

**Comment by vaibhavparatwar** [comment]
https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/9882/console
15N deployment also passed with this PR.

**Pull request merged**